cpp/src/utilities/column_utils.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Eyal Rozenberg <eyalroz@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "column_utils.hpp"
+#include <utilities/error_utils.hpp>
+
+// This should go up into more general utility code...
+static constexpr inline bool logical_xor(bool x, bool y) { return (x and (not y)) or ((not x) and y) ; }
+
+namespace cudf {
+
+/**
+ * @brief Ensures a gdf_column is valid, i.e. that its fields are consistent
+ * with each other, and logical in themselves, in representing a proper column.
+ */
+void validate(const gdf_column& column)
+{
+    CUDF_EXPECTS ( (column.data != nullptr or column.size == 0)          , "Supposedly non-empty column is missing its data");
```

Comment:
```suggestion
    CUDF_EXPECTS ( (column.data != nullptr or column.size == 0)          , "Column size is non-zero but data pointer is null");
```

Reply:
The caller of this code does not need to know about the "data pointer". It's similar to "non-zero size"; I don't want to mention I have a size field. Minimum amount of necessary information...

Reply:
I don't want imprecise language like "supposedly" in error messages.